Sources/Sentry/SentryClient.m

```diff
@@ -568,30 +569,29 @@ - (SentryEvent *_Nullable)callEventProcessors:(SentryEvent *)event
 
 - (void)setSdk:(SentryEvent *)event
 {
-    // Every integration starts with "Sentry" and ends with "Integration". To keep the payload of
-    // the event small we remove both.
-    NSMutableArray<NSString *> *integrations = [NSMutableArray new];
-    for (NSString *integration in self.options.enabledIntegrations) {
-        NSString *withoutSentry = [integration stringByReplacingOccurrencesOfString:@"Sentry"
-                                                                         withString:@""];
-        NSString *trimmed = [withoutSentry stringByReplacingOccurrencesOfString:@"Integration"
-                                                                     withString:@""];
-        [integrations addObject:trimmed];
-    }
-
-    NSMutableDictionary *sdk = @{
-        @"name" : SentryMeta.sdkName,
-        @"version" : SentryMeta.versionString,
-        @"integrations" : integrations
+    if (event.sdk) {
+        return;
     }
-                                   .mutableCopy;
 
-    if (nil == event.sdk) {
-        if (event.extra[@"__sentry_sdk_integrations"]) {
-            [sdk setValue:event.extra[@"__sentry_sdk_integrations"] forKey:@"integrations"];
+    id integrations = event.extra[@"__sentry_sdk_integrations"];
+    if (!integrations) {
+        integrations = [NSMutableArray new];
+        for (NSString *integration in self.options.enabledIntegrations) {
+            // Every integration starts with "Sentry" and ends with "Integration". To keep the
+            // payload of the event small we remove both.
+            NSString *withoutSentry = [integration stringByReplacingOccurrencesOfString:@"Sentry"
+                                                                             withString:@""];
+            NSString *trimmed = [withoutSentry stringByReplacingOccurrencesOfString:@"Integration"
+                                                                         withString:@""];
+            [integrations addObject:trimmed];
         }
-        event.sdk = sdk;
     }
+
+    event.sdk = @{
+        @"name" : self.options.sdkInfo.name,
```

Comment:
The value from SentryMeta is added to envelope headers so Relay can also know what SDK is sending the data to its backend. This can help us troubleshoot issues where one SDK is sending bad data.

Ideally we find a way to change that value and affect all corners of the SDK, instead of reporting a different value here and on the envelope header, for example.